plugin-desktopswitch/desktopswitch.cpp

```diff
@@ -46,8 +46,9 @@ DesktopSwitch::DesktopSwitch(const ILxQtPanelPluginStartupInfo &startupInfo) :
     ILxQtPanelPlugin(startupInfo),
     m_pSignalMapper(new QSignalMapper(this)),
     m_desktopCount(KWindowSystem::numberOfDesktops()),
-    mRows(1),
-    mDesktops(new NETRootInfo(QX11Info::connection(), NET::NumberOfDesktops | NET::CurrentDesktop | NET::DesktopNames, NET::WM2DesktopLayout))
+    mRows(settings()->value("rows", 1).toInt()),
```

Comment:
IMHO it's a good practice to read the configuration only in one place (avoiding typos in keys, future changes more straight forward, etc..) . Therefore everything was set only in `settingsChanged``